python/pyspark/sql/streaming/query.py

```diff
@@ -189,6 +226,11 @@ def recentProgress(self) -> List[Dict[str, Any]]:
         The number of progress updates retained for each stream is configured by Spark session
         configuration `spark.sql.streaming.numRecentProgressUpdates`.
 
+        Returns
+        -------
+        str or dict or list
+            An array of the most recent :class:`StreamingQueryProgress` updates for this query.
```

Comment:
Should probably mention that return a dictionary instead of `StreamingQueryProgress` instance.

Reply:
I choose to keep `StreamingQueryProgress` and add `dict` here. Because `StreamingQueryProgress` can help user to understand how it is in `dictionary`.